### DISCUSSION TITLE
Fullscreen notifications when meeting is close

### DISCUSSION THREAD

**Discussion by thejens** [pakerwreah/Calendr#283] [Ideas]
Hey! Just found this app after considering building it myself (though as I don't know Swift yet, using yours is a ton easier) . 

I'm only missing one feature, and if I knew how to code it I'd contribute it myself. 

It'd be awesome if the menu bar item, counting down to the next event, got a bit more aggressive when the event nears. Essentially I am looking for something that'd break my flow-state and tell me to run off to the next meeting.

For instance it could turn red, flash somehow, blink an icon or do something else to grab attention.  perhaps configurable if that behaviour should start 10, 5 or 1 minute before the meeting starts.

Regardless of above, great app!

**Comment by pakerwreah** [comment]
Hey! Welcome to the project :)

This feature was requested a couple of times before.
Genuine question: why is that any different than using the native Calendar.app alerts / banners?

**Comment by thejens** [comment]
Banner fatigue I guess (calendar, slack, mail, messages etc. all push banners), They're physically in the way, so will either have them disabled due to focus mode , or often close them on "autopilot" , and then N minutes later forgot about them

**Comment by davidashman** [comment]
Have you considered adding functionality like InYourFace?  I end up using this and InYourFace to really let me know something is coming up.  I briefly looked at adding that type of thing to Calendr but my experience on native macOS apps is very limited.

**Comment by pakerwreah** [comment]
(sorry, long text ahead)

That is one of the things in FAQ that says that I will (probably) never do.

It may look like a simple feature to add, but people will have different needs and I will have to create a whole UI and a bunch of new settings for it.

It's somewhat related to other most requested feature I get, which is event/reminder creation. It generates a lot of maintenance and it's not the purpose of this app.

Even if I implement this, you can still dismiss the fullscreen notification and forget about your meeting anyway. I truly believe it's not the kind of problem a software can solve.

Regarding the new "grab attention" feature I added to the latest version, I didn't want to do it. Apple already has a nice sticky Calendar notification, which has configurable sound and "snooze" built-in, or even some specialized app like the one you mentioned.

I only did it because it was easy enough to implement and it's not really replacing the native notifications, but adding a non-intrusive layer to help you not forget about your appointments. That way I can use it myself without interfering with my workflow.

I didn't even add a settings to disable it, which I think people will certainly ask for soon, which will generate more maintenance, and so on...

Don't get me wrong, I'm not trying to be an asshole by refusing to implement certain things. I'm just trying not to turn this app into a Swiss Army Knife, because it will be impossible to maintain. Maybe in the future if the app grows enough and I get more people involved, and more PRs coming in, we can all work on this together.

Good opportunity for you guys to try some Swift 😅

I hope I was able to make myself clear without being rude.

I'm gonna move this issue to a discussion, so it gets more visibility.

**Reply by PuzzledUser** [reply]
Aha, I enjoyed your long text! But came here because I didn't like the distracting flashing RED in my menu bar. You can't please all the people all the time!? Please add a switch to turn it off. 

By the way, you didn't add a beep as well did you? I noticed a funny noise as I was preparing for a recent meeting and didn't have time to look for what was doing it (but I did see red)! 😉

**Reply by pakerwreah** [reply]
yep, I added both of those annoyances.
I also disabled them by default in [v1.14.7](https://github.com/pakerwreah/Calendr/releases/tag/v1.14.7)
There are new settings now to turn them on.

**Reply by PuzzledUser** [reply]
Much appreciated Carlos. Thank you for your simple and effective Calendr. (I do like the up next feature and think you've found a clever balance between features and simplicity)❣️

**Reply by pakerwreah** [reply]
Thank you!
That means a lot to me 🙂

**Comment by pakerwreah** [comment]
I just noticed that there was already a discussion around it here:
- https://github.com/pakerwreah/Calendr/discussions/248
